### ISSUE TITLE
CKAN 2.10 - Authenticated catalog users not redirected to `catalog-admin`

### ISSUE THREAD

**Issue by nickumia-reisys** [GSA/data.gov#4371]
## How to reproduce

1. Try to log in to [catalog-dev](https://catalog-dev-admin-datagov.app.cloud.gov/dataset/)

## Expected behavior

Login successful.

## Actual behavior

SAML doesn't work..
![image](https://github.com/GSA/data.gov/assets/85196563/fe7df92d-1955-49aa-8d61-cb193e14e02c)

## Sketch

- [ ] Figure out how to redirect users to `catalog-admin` if logged in
  - The old logic for doing this broke because CKAN removed the `auth_tkt` cookie.

**Comment by nickumia-reisys** [comment]
While debugging with the team yesterday, we found out the real issue is that the `auth_tkt` cookie was removed in CKAN 2.10.  Login works and has no issues.  Our `catalog-proxy` redirects requests to `catalog-admin` if a user is logged in; otherwise, it redirects to `catalog-web`.  The logic was controlled by the `auth_tkt` cookie.  Since it no longer exists, after login, the user is redirected to `catalog-web` who didn't authenticate the user (`catalog-admin` has the login session).  The issue is that the user can't access the server they are logged into.

@FuhuXia proposed a solution where we can possibly replicate the old logic by letting `catalog-proxy` manage the user-logged-in cookie.

According to this ticket, the change may only be implemented and take effect on `CKAN 2.10.1`, but we tried to use `CKAN 2.10.0` and it still had the same issue...  so there is no fallback option if we don't have a solution to do proper user redirecting
- https://github.com/ckan/ckan/issues/7406#issuecomment-1438608208
